test/promise.cc

```diff
@@ -0,0 +1,29 @@
+#include "napi.h"
+
+using namespace Napi;
+
+Value IsPromise(const CallbackInfo& info) {
+  return Boolean::New(info.Env(), info[0].IsPromise());
+}
+
+Value ResolvePromise(const CallbackInfo& info) {
+  auto resolver = Promise::Resolver::New(info.Env());
```

Comment:
I don't feel too strongly about this, but what about `s/Resolver/Deferred/g`? The only reason I mention it is that this makes things look too much like V8. IMO using "Deferred" distances us from the engine.That's also why I used "deferred" in the C API.

Reply:
That makes sense. I did copy `Resolver` from v8.